packages/element/package.json

```diff
@@ -0,0 +1,30 @@
+{
+	"name": "@wordpress/element",
+	"version": "0.0.1",
+	"description": "Element React module for WordPress",
+	"author": "WordPress",
+	"license": "GPL-2.0-or-later",
+	"keywords": [
+		"wordpress",
+		"element",
+		"react"
+	],
+	"homepage": "https://github.com/WordPress/gutenberg/tree/master/packages/element/README.md",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/WordPress/gutenberg.git"
+	},
+	"bugs": {
+		"url": "https://github.com/WordPress/gutenberg/issues"
+	},
+	"main": "src/index.js",
+	"dependencies": {
+		"@wordpress/is-shallow-equal": "1.0.2",
+		"lodash": "4.17.5",
+		"react": "16.3.2",
+		"react-dom": "16.3.2"
```

Comment:
Does this mean, we can remove `react` and `react-dom` from the root level package.json?

Reply:
Technically we can, but they won't be tracked in `package-json.lock`. In the previous PR we decided to keep them in sync with the root level `package.json`. We can revisit later. Personally, I would prefer to have it working with lock files and have only one place to include them.

Reply:
I think it's not really important since it's just for test. But I agree it's a concern for other dependencies that are bundled and I think we should have a separate lock per module.

Reply:
You are right, I forgot about it. As far as I remember, those local lock files, don't really work when using `lerna bootstrap`. @ntwb - or do they and we disabled to follow semantic versioning?

Reply:
Hmmm, I've not got an accurate recollection, there was some trouble I think.

That said, https://github.com/WordPress/packages do not have lock files, why would the https://github.com/WordPress/gutenberg require lock files.

tl;dr: Lock files are for apps, not packages or modules

Reply:
I found another option which should work, but it would create lots of copies of popular dependencies like `lodash`. See: https://github.com/lerna/lerna/issues/856#issuecomment-375838518:
`lerna exec -- npm i` - then every package would use its own lock file for conistency.

Reply:
Maybe we should give it a try and have the lock file only for the root `package.json` because Gutenberg is an app 😄

Reply:
I understand your point but I think I disagree. The difference between "app" and "packages" is superficial. What's the app when your whole app is a collection of packages? How do you guarantee an npm install won't inadvertently break your application?

Reply:
You can't which is one of my points in that if you look down an "app" or "package" with specific package versions using a lock file then what is the point of SemVer, you may as well release everything as a _major_ release and never worry about _patch_ or _minor_ releases for an "app" or "package" that is depended upon

Reply:
So you're ok with merging a PR and than your automatic CI job releasing your packages/app break because it go the wrong package you never tested?

I'm not. I see where you're coming from but that's true when dependencies get bundled together to build a single bundle which is not the case in our setup, each package is an app that generates its own script. We need predictability.

Reply:
On a "package" side of things I am happy to let SemVer do what it was designed to do.

On an "app" I'm with you 100% and everything should be locked down.

Anything in between is tricky, which is where we are at here...

If the [Gutenberg packages](https://github.com/WordPress/gutenberg/tree/master/packages) are released using [SemVer](https://semver.org/) then using an explicit locked down version e.g. `1.0.0` and not `~1.0.0` or `^1.0.0` of that package for Gutenberg **with** a lock file will work perfectly for Gutenberg as the only time a dependancy is bumped is when it is manually bumped which would include testing that the version bump works as expected.

Individual packages are then free to be used in the wider community and will continue to adhere to SemVer allowing other packages to depend upon them and update as they see fit per their own SemVer policy.

Anyways, that's my take, should probably save this discussion for the follow up PR @gziolo has planned 😄

Reply:
But I'm still not sure I follow you. There's no difference between an "app" and a "package". All Gutenberg packages are "apps" because they're all released as such as WordPress scripts.

Gutenberg will never "require" or depend on Gutenberg packages because they're built in the same repository. That may change (not certain) when the merge happens but until then, all the packages are apps, there's no higher-level package-lock.json where they can be locked.